Extensions/CopyStageVariableToReleaseVariable/Src/Tasks/CopyAgentJobVariableToReleaseVariable/copyvartorelease.ts

```diff
@@ -0,0 +1,64 @@
+import tl = require('./node_modules/azure-pipelines-task-lib/task');
+const azdev = require('azure-devops-node-api'); 
+import * as ReleaseApi from './node_modules/azure-devops-node-api/ReleaseApi';
+import * as ReleaseInterfaces from './node_modules/azure-devops-node-api/interfaces/ReleaseInterfaces';
+
+async function run() {
+    try {
+        var stageVariableNamesArray = tl.getDelimitedInput("stageVariableNames", ',', true);
+        var releaseVariableNames = tl.getInput("releaseVariableNames", false);
+        var releaseVariableNamesArray : string[] = [];
+        if (!!releaseVariableNames)
+        {
+            releaseVariableNamesArray = releaseVariableNames.split(',');
+            if(releaseVariableNamesArray.length != stageVariableNamesArray.length)
+            {
+                throw new Error(tl.loc("ReleaseVarLengthShoulbeSameAsStageVarLength"));
+            }
+        }
+        else
+        {
+            releaseVariableNamesArray = stageVariableNamesArray;
+        }
+
+        var collectionUri = tl.getVariable("System.CollectionUri");
+        var teamProject = tl.getVariable("System.TeamProject");
+        var releaseId = tl.getVariable("Release.ReleaseId");
+
+        const accessToken: string = tl.getEndpointAuthorizationParameter('SYSTEMVSSCONNECTION', 'ACCESSTOKEN', false);
+
+        let authHandler = azdev.getHandlerFromToken(accessToken);
+        let connection = new azdev.WebApi(collectionUri, authHandler);    
+        let releaseApiObject: ReleaseApi.IReleaseApi  = await connection.getReleaseApi();
+
+        tl.debug("Get release");
+        var release: ReleaseInterfaces.Release = await releaseApiObject.getRelease(teamProject, releaseId);
```

Comment:
try catch? else in case of exception you don't see "FailedToFetchRelease" error.